uaclient/cli.py

```diff
@@ -1477,16 +1518,24 @@ def wrapper(*args, **kwargs):
                     else:
                         msg_tmpl = ua_status.LOG_CONNECTIVITY_ERROR_TMPL
                     logging.exception(msg_tmpl.format(**msg_args))
-                print(ua_status.MESSAGE_CONNECTIVITY_ERROR, file=sys.stderr)
+
+                event.error(error_msg=ua_status.MESSAGE_CONNECTIVITY_ERROR)
+                event.info(
+                    info_msg=ua_status.MESSAGE_CONNECTIVITY_ERROR,
+                    file_type=sys.stderr,
+                )
             lock.clear_lock_file_if_present()
+            event.process_events()
             sys.exit(1)
         except exceptions.UserFacingError as exc:
             with util.disable_log_to_console():
                 logging.error(exc.msg)
-            print("{}".format(exc.msg), file=sys.stderr)
+            event.error(error_msg=exc.msg)
+            event.info(info_msg="{}".format(exc.msg), file_type=sys.stderr)
             if not isinstance(exc, exceptions.LockHeldError):
                 # Only clear the lock if it is ours.
                 lock.clear_lock_file_if_present()
+            event.process_events()
             sys.exit(exc.exit_code)
         except Exception:
```

Comment:
In case of unexpected exception, should we return then through json or assume that the command should fail just like the cli one ?

Reply:
I think we should always return with the json